### PR TITLE
Remove leaks from snapshot_creator

### PR DESCRIPTION
**Pull request by ry**

### CODE REVIEW

File: libdeno/snapshot_creator.cc [hunk 1]
```diff
@@ -40,5 +40,9 @@ int main(int argc, char** argv) {
   std::ofstream file_(snapshot_out_bin, std::ios::binary);
```
Comment (F001) [inline libdeno/snapshot_creator.cc]: I don't understand why we clone the snapshot into a `std::string` before writing to file. This is a waste. (And also, there is a risk: `std::string` assumes  no `\0` in the middle of array. )
Why not remove `snapshot_str` and use `file_.write(snapshot.data_ptr, snapshot.data_len);` directly?

Comment (piscisaureus) [reply libdeno/snapshot_creator.cc]: > `std::string` assumes no `\0` in the middle of array.

I don't think that's true per se (it's only assumed when initializing the std::string from a c string).

Comment (ry) [reply libdeno/snapshot_creator.cc]: @F001 Fixed - thanks.

File: .travis.yml [hunk 2]
```diff
@@ -93,13 +93,15 @@ script:
...
+  # LSAN build. We are in the process of getting a completely clean LSAN build,
+  # but it will take some work. So for now we just run a subset of the tests.
   echo is_asan=true >> target/debug/args.gn
   echo is_lsan=true >> target/debug/args.gn
-  DENO_BUILD_MODE=debug ./tools/build.py -j2 test_cc
-  ASAN_OPTIONS=detect_leaks=1 ./target/debug/test_cc
+  # We want to detect leaks during the build process as well as when executing
+  # the tests. So set the ASAN_OPTIONS env var before build.py is run.
+  export ASAN_OPTIONS=detect_leaks=1
+  DENO_BUILD_MODE=debug ./tools/build.py -j2
```
[the remark this answers is not in the document]
Comment (ry) [reply .travis.yml]: I agree that's better. Fixed.

**Pull request merged**